java/compaction/compaction-job-execution/src/main/java/sleeper/compaction/jobexecution/CompactSortedFilesRunner.java

```diff
@@ -286,18 +264,18 @@ public static void main(String[] args)
         }
 
         ObjectFactory objectFactory = new ObjectFactory(instanceProperties, s3Client, "/tmp");
-        CompactSortedFilesRunner runner = new CompactSortedFilesRunner(
-                instanceProperties, objectFactory,
-                tablePropertiesProvider,
-                propertiesReloader,
-                stateStoreProvider,
-                jobStatusStore,
-                taskStatusStore,
-                UUID.randomUUID().toString(),
-                sqsJobQueueUrl,
-                sqsClient,
-                ecsClient,
-                type);
+        CompactSortedFilesRunner runner = CompactSortedFilesRunner.builder()
+                .instanceProperties(instanceProperties)
+                .objectFactory(objectFactory)
+                .tablePropertiesProvider(tablePropertiesProvider)
+                .propertiesReloader(propertiesReloader)
+                .stateStoreProvider(stateStoreProvider)
+                .jobStatusStore(jobStatusStore)
+                .taskStatusStore(taskStatusStore)
+                .taskId(UUID.randomUUID().toString())
+                .sqsJobQueueUrl(sqsJobQueueUrl)
+                .type(type)
+                .buildWithDefaultClients();
```

Comment:
The SQS and ECS clients are still created and shut down in this method, but are never used.